tc/src/main/java/com/epam/tc/dto/UserDTO.java

```diff
@@ -24,10 +24,15 @@ public boolean canSubscribe(Course course) {
     }
 
     public boolean canSee(Course course) {
+        if (course.getStatus().getName().equals("Proposal")) {
+            return !((user == null) || (user.getUserRole().equals("User")));
```

Comment:
In theory, if you all did well, then there shouldn't be such checks: user == null